### PR TITLE
api created and tested

### LINKED ISSUE #80: Warden Cancel gatepass API
Make an API to cancel the gatepass

### PR DESCRIPTION
**Pull request by karamarora20**
put request api created

### CODE REVIEW

File: client-side/src/Screens/Login/auth.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ export default function Auth() {
       .then((response) => {
         Cookies.set("ACCESS_TOKEN", response.ACCESS_TOKEN);
         const access_token = response.ACCESS_TOKEN;
+        console.log(access_token);
```
Comment (palashmunshi) [inline client-side/src/Screens/Login/auth.js:31]: remove this console log

File: server-side/node-mssql-crud/src/database/query.js [hunk 2]
```diff
@@ -93,6 +93,7 @@ export const queries = {
 
   acceptGatepass: "UPDATE [gps_db].[gps_db].[gps_gatepassmaster] SET status='Approved' WHERE request_id=@id;",
   rejectGatepass: "UPDATE [gps_db].[gps_db].[gps_gatepassmaster] SET status='Rejected' WHERE request_id=@id;",
+  cancelApprovedGatepass:"UPDATE [gps_db].[gps_gatepassmaster] SET status = 'cancelled', approved_or_rejected_by= @user_id , approved_or_rejected_date = @date, approved_or_rejected_time = @time, comments = @comments WHERE request_id= @req_id",
```
Comment (palashmunshi) [inline server-side/node-mssql-crud/src/database/query.js]: 1. do we update approved_or_rejected_date / time when we cancel the gatepass?? I think it is only updated when gatepass is approved or cancelled. same for approved_or_rejected_by. Please tell.

File: server-side/node-mssql-crud/src/controllers/warden.controller.js [hunk 3]
```diff
@@ -22,4 +22,27 @@ export const gatepassApproveOrReject = async (req, res) => {
...
+        console.log(req_id);
+        console.log(comments);
+        const pool = await getConnection();
+        const result = await pool
+          .request()
+          .input("user_id", sql.VarChar, id)
+          .input("req_id",sql.Int,req_id)
+          .input("comments",sql.VarChar,comments)
+          .input("date",sql.VarChar,applied_date)
+          .input("time",sql.Time,d)
```
Comment (palashmunshi) [inline server-side/node-mssql-crud/src/controllers/warden.controller.js]: what is this time please show a screenshot of the database by today as it is not working in my system.

**Review by palashmunshi — CHANGES_REQUESTED** [review]

**Comment by palashmunshi** [comment]
After all changes, pull the latest version of localFixed and then merge localFixed into your branch. You'll get conflicts, you'll have to resolve those and merge then push.

File: client-side/src/Screens/Login/auth.js (hunk 1 above)
Comment (karamarora20) [reply client-side/src/Screens/Login/auth.js:31]: Sry got left will do right away

**Review by palashmunshi — APPROVED** [review]
nice

**Pull request merged**